File: py/test-poll.py
#
#  Copyright (c) 2014,2015 NetEase, Inc. and other Pomelo contributors
#  MIT Licensed.
#

import pomelo
import time
import os

def lc_callback(*args):
    if args[0] == Client.PC_LOCAL_STORAGE_OP_WRITE:
        lcstr = args[1]

        with open("pomelo.dat", "w") as f:
            f.write(lcstr)
        return 0

    else:
        if os.path.exists("pomelo.dat"):
            with open("pomelo.dat", "r") as f:
               return f.read()
        else:
            return None


def resp_callback(rc, resp):
    print 'request status: ', Client.rc_to_str(rc), 'resp:', resp

def notify_callback(rc):
    print 'notify status: ', Client.rc_to_str(rc)

def event_callback(ev, arg1, arg2):
    if ev == Client.PC_EV_USER_DEFINED_PUSH:
        print 'get push message, route: ', arg1, 'msg: ', arg2
    else:
        print 'network event:', Client.ev_to_str(ev), arg1, arg2

Client = pomelo.Client

Client.lib_init(Client.PC_LOG_WARN, None, None)

c = Client()

# disable tls, enable poll
c.init(False, True, lc_callback)

handler_id = c.add_ev_handler(event_callback)

c.connect('127.0.0.1', 3010)

time.sleep(1);

c.request('connector.entryHandler.entry', '{"name": "test"}', 10, resp_callback);

c.notify('test.testHandler.notify', '{"content": "test content"}', 10, notify_callback)

time.sleep(10)

c.poll()

c.rm_ev_handler(handler_id)

time.sleep(10)

ret = c.destroy()

Client.lib_cleanup()

File: java/com/netease/pomelo/TestPoll.java
/**
 * Copyright (c) 2014 NetEase, Inc. and other Pomelo contributors
 * MIT Licensed.
 */

package com.netease.pomelo;

import com.netease.pomelo.Client;

public class TestPoll {
  static public void main(String [] args) {

    Client.libInit(Client.PC_LOG_WARN, null, null);

    Client c = new Client();

    // tls, polling
    c.init(false, true, new Client.LocalStorage() {
      public int write(String s) {return -1; }
      public String read() {return null;}
    });

    c.connect("127.0.0.1", 3010);

    c.addEventHandler(Client.PC_EV_USER_DEFINED_PUSH, "onPush", new Client.EventHandler() {
        public void handle(int ev_type, String arg1, String arg2) {
            System.out.println("get push message: " + arg2);
        }
    });

    c.request("connector.entryHandler.entry", "{\"name\": \"test\"}", 10, new Client.RequestCallback() {
        public void handle(int rc, String resp) {
            System.out.println("get resp:" + resp);
        }
    });

    c.notify("test.testHandler.notify", "{\"content\": \"test\"}", 10, new Client.NotifyCallback() {
        public void handle(int rc) {
            System.out.println("get status:" + Client.rcToStr(rc));
        }
    });

    try {
        Thread.sleep(10000);
    } catch (Exception e) {

    }

    c.poll();

    try {
        Thread.sleep(10000);
    } catch (Exception e) {

    }
    c.destroy();
    Client.libCleanup();
  }
}
